spring-cloud-azure-markdown-appdendix-generator/src/main/java/com/azure/spring/cloud/internal/Main.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2012-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.azure.spring.cloud.internal;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+import org.springframework.util.StreamUtils;
+
+/**
+ * Generate markdown files with configuration properties based on the path of the file to
+ * be output and the pattern included in the properties.
+ */
+public final class Main {
+
+	private Main() {
+	}
+
+	/**
+	 * When the parent dictionary is found, change the suffix of the output file to md and
+	 * remove the "_" from the filename, generate configuration properties markdown files.
+	 * @param args arg[0] the output file, arg[1] the properties inclusion pattern
```

Comment:
```
This method is used to ...

Arguments:
1. args[0]: output file name. The file name must start with... And in actual outputted file name, ".adoc" will be replaced by ".md", "_" will be replace by " ". For example: ...  This is required because ...
2. args[1]: 
```